Vic2ToHoI4/Source/HOI4World/HoI4Country.cpp

```diff
@@ -874,4 +874,71 @@ std::optional<HoI4FocusTree> HoI4::Country::getNationalFocus() const
 	{
 		return nullopt;
 	}
+}
+
+void HoI4::Country::addSphereling(std::string Sphereling)
+{
+	double defaultAutonomy = 0.5;
+	spherelings.insert(make_pair(Sphereling, defaultAutonomy));
+}
+
+double HoI4::Country::getSpherelingAutonomy(std::string spherelingTag) const
+{
+	auto sphereling = spherelings.find(spherelingTag);
+	if (sphereling == spherelings.end())
+	{
+		LOG(LogLevel::Debug) << "Could not find " << sphereling->first << " in spherelings";
+		return 0.5;
+	}
+	else
+	{
+		return sphereling->second;
+	}
+}
+
+void HoI4::Country::setSpherelingAutonomy(std::string spherelingTag, double autonomy)
+{
+	auto sphereling = spherelings.find(spherelingTag);
+	if (sphereling == spherelings.end())
+	{
+		LOG(LogLevel::Debug) << "Could not find " << sphereling->first << " in spherelings";
+		sphereling->second = 0.5;
+	}
+	else
+	{
+		sphereling->second = autonomy;
+	}
+}
+
+double HoI4::Country::calculateInfluenceFactor()
+{
+	auto GPInfluences = getGPInfluences();
+	if (sphereLeader.empty())
+	{
+		LOG(LogLevel::Debug) << tag << " has no sphere leader set";
+		return 55.5;
+	}
+	else
+	{
+		double influenceFactor = 0;
+		for (auto& influenceItr: GPInfluences)
+		{
+			if (influenceItr.first != sphereLeader)
+			{
+				influenceFactor += influenceItr.second;
+			}
+			if (influenceItr.first == sphereLeader)
+			{
+				influenceFactor -= 1.5 * influenceItr.second;
+			}
+		}
+		//1 is used because if freedom_level is too close to 0 (how close?)
+		//it's displayed ingame as being halfway: 0.5 instead of 0.0000something
+		double minInfluenceFactor = 1;
```

Comment:
If you include algorithm, you can just do
```return std::clamp(influenceFactor, 1, 100);```

Reply:
Aah, yes. Clamp is the word I've been looking for. Just ran out of RAM while doing it :D :D :D

Reply:
`Severity	Code	Description	Project	File	Line	Suppression State	Detail Description
Error (active)	E0304	no instance of overloaded function "std::clamp" matches the argument list	Vic2ToHoI4	D:\source\repos\Vic2ToHoI4\Vic2ToHoI4\Source\HOI4World\HoI4Country.cpp	922		            argument types are: (double, int, int)
`
`return std::clamp(influenceFactor, 1.0, 100.0);` removed the error, though.

Reply:
Yeah, it's templated, so all inputs will have to be the same type. Was that the full error you got? Or was that at the end of dozens of lines of nonsense. If it was the full thing, it means some improvements to Visual Studio have dropped.

Reply:
There were 4 others related to clamp like`Severity	Code	Description	Project	File	Line	Suppression State
Error	C2672	'std::clamp': no matching overloaded function found	Vic2ToHoI4	D:\source\repos\Vic2ToHoI4\Vic2ToHoI4\Source\HOI4World\HoI4Country.cpp	935	
`